src/coloring.jl

```diff
@@ -543,10 +560,32 @@ function postprocess!(
     end
     if star_or_tree_set isa StarSet
         # only the colors of the hubs are used
-        (; hub) = star_or_tree_set
+        (; hub, spokes) = star_or_tree_set
+        nb_trivial_star = 0
+
+        # Iterate through all non-trivial stars
         for s in eachindex(hub)
             j = hub[s]
-            color_used[color[j]] = true
+            if j > 0
+                color_used[color[j]] = true
+            else
+                nb_trivial_star += 1
+            end
+        end
+
+        # Process the trivial stars (if any)
+        if nb_trivial_star > 0
+            for s in eachindex(hub)
+                j = hub[s]
+                if j < 0
+                    i = spokes[s][1]
+                    j = abs(j)
+                    if !color_used[color[i]] && !color_used[color[j]]
```

Comment:
@amontoison I think the problem is here. During decompression, we always recover coefficients based on the color of the star's hub. With this change, if `color_used[color[i]]`, we're not assigning any color to `j` at all. Either we revert this or we adapt `symmetric_coefficient` accordingly. I'll try adapting.

Reply:
The hub swapping seemed to solve it

Reply:
Good catch Guillaume!

We probably need to a similar modification with the trivial trees (when we will handle them).